pkg/controller/deployment/util/deployment_util.go

```diff
@@ -787,6 +787,17 @@ func GetActualReplicaCountForReplicaSets(replicaSets []*extensions.ReplicaSet) i
 	return totalActualReplicas
 }
 
+// GetReadyReplicaCountForReplicaSets returns the number of ready pods corresponding to the given replica sets.
+func GetReadyReplicaCountForReplicaSets(replicaSets []*extensions.ReplicaSet) int32 {
```

Comment:
No, none of the deployment utils handles duplicate replica sets (cache Lists should already return deduped results).